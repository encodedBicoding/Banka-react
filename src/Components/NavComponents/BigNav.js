import React, { useEffect } from 'react';
import { Link, NavLink } from 'react-router-dom';
import { connect } from 'react-redux';

export const BigNav = (
  { home, 
    about, 
    contact, 
    signup, 
    login, 
    showLoginFormModal, 
    showSignupFormModal }) => {
  return (
    <div className="nav">
      <div className="nav_logo">
        <span className="nav_title">B</span>
        <span className="nav_title">A</span>
        <span className="nav_title">N</span>
        <span className="nav_title">K</span>
        <span className="nav_title nav_last_child">A</span>
        <p className="nav_slogan">...first online mobile banking service</p>
      </div>
      <nav className="home_nav">
        <ul className="home_nav_header">
          <li className="home_nav_list">
            <Link to="/" id="nav_link">
              {home}
            </Link>
          </li>
          <li className="home_nav_list">
            <a href="#about" id="nav_link">{about}</a>
          </li>
          <li className="home_nav_list">
            <a href="#contact" id="nav_link"> {contact}</a>
          </li>
          <li className="home_nav_list" datatest="signupmodal" onClick={showSignupFormModal}>
            {signup}
          </li>
          <li className="home_nav_list" onClick={showLoginFormModal}>
            {login}
          </li>
        </ul>
      </nav>
    </div>
  );
};

const mapDispatchToProps = dispatch => ({
  showSignupFormModal: () => dispatch({ type: 'SIGNUP_MODAL'}),
  showLoginFormModal: () => dispatch({ type: 'LOGIN_MODAL'})
})
export default connect(null, mapDispatchToProps)(BigNav);
