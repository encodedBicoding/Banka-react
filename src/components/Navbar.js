import React from "react";
import { Link } from "react-router-dom";

const Navbar = props => {
  return (
    <React.Fragment>
      <div className="nav_logo">
        <span className="nav_title">B</span>
        <span className="nav_title">A</span>
        <span className="nav_title">N</span>
        <span className="nav_title">K</span>
        <span className="nav_title nav_last_child">A</span>
        <p className="nav_slogan">...first online mobile banking service</p>
      </div>
      <nav className="home_nav">
        <ul className="home_nav_header">
          <li className="home_nav_list">
            <Link to="/" id="nav_link">
              {props.home}
            </Link>
          </li>
          <li className="home_nav_list">
            <Link to="#about" id="nav_link">
              {props.about}
            </Link>
          </li>
          <li className="home_nav_list">
            <Link to="#contact" id="nav_link">
              {props.contact}
            </Link>
          </li>
          <li className="home_nav_list">{props.signup}</li>
          <li className="home_nav_list">{props.login}</li>
        </ul>
      </nav>
    </React.Fragment>
  );
};

export { Navbar };
