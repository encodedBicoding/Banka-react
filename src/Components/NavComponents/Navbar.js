import React, { useEffect, useState } from 'react';
import BigNav from './BigNav.js';
import SmallNav from './SmallNav.js';
import './NavStyle.css';

const Navbar = props => {
  let [scrollPosition, setScrollPosition] = useState(0);
  useEffect(() => {
    window.onscroll = () => {
      setScrollPosition(document.documentElement.scrollTop);
    };
  }, [scrollPosition]);
  return (
    <div className="main-container">
      {scrollPosition > 20 ? (
        <SmallNav
          home={props.home}
          about={props.about}
          contact={props.contact}
          signup={props.signup}
          login={props.login}
        />
      ) : (
        <BigNav
          home={props.home}
          about={props.about}
          contact={props.contact}
          signup={props.signup}
          login={props.login}
        />
      )}
    </div>
  );
};

export default Navbar;
