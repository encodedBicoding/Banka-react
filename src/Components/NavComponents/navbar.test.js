import React from 'react';
import Navbar from './Navbar';
import { shallow } from 'enzyme';
import { BigNav } from './BigNav';

describe('Handle Navbar Component test', () => {
  it('Should render correctly', () => {
    const wrapper = shallow(<Navbar />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('test bignav mount', () => {
  let wrapper, props, instance;
  beforeEach(() => {
    props = {
      showSignupFormModal: jest.fn(),
      showLoginFormModal: jest.fn(),
    }
    wrapper = shallow(<BigNav {...props}/>);
    instance= wrapper.instance()
  })
  it('should mount properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should dispatch signup modal', () => {
    const showSignupFormModal =jest.fn();
    const link = wrapper.find('li[datatest="signupmodal"]');
    link.simulate('click');
    showSignupFormModal();
    expect(showSignupFormModal).toHaveBeenCalled();
  })
  it('should test useEffect function', () => {
    const window = {
      onscroll: jest.fn()
    }
    const scrollPosition = 0;
    const setScrollPosition = jest.fn();
    window.onscroll()
    setScrollPosition()
    expect(scrollPosition).toMatchSnapshot()
    expect(setScrollPosition).toHaveBeenCalled();
  })
});

