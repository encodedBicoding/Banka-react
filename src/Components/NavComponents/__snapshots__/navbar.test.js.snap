// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Handle Navbar Component test Should render correctly 1`] = `
<div
  className="main-container"
>
  <Connect(BigNav) />
</div>
`;

exports[`test bignav mount should mount properly 1`] = `
<div
  className="nav"
>
  <div
    className="nav_logo"
  >
    <span
      className="nav_title"
    >
      B
    </span>
    <span
      className="nav_title"
    >
      A
    </span>
    <span
      className="nav_title"
    >
      N
    </span>
    <span
      className="nav_title"
    >
      K
    </span>
    <span
      className="nav_title nav_last_child"
    >
      A
    </span>
    <p
      className="nav_slogan"
    >
      ...first online mobile banking service
    </p>
  </div>
  <nav
    className="home_nav"
  >
    <ul
      className="home_nav_header"
    >
      <li
        className="home_nav_list"
      >
        <Link
          id="nav_link"
          to="/"
        />
      </li>
      <li
        className="home_nav_list"
      >
        <a
          href="#about"
          id="nav_link"
        />
      </li>
      <li
        className="home_nav_list"
      >
        <a
          href="#contact"
          id="nav_link"
        >
           
        </a>
      </li>
      <li
        className="home_nav_list"
        datatest="signupmodal"
        onClick={[MockFunction]}
      />
      <li
        className="home_nav_list"
        onClick={[MockFunction]}
      />
    </ul>
  </nav>
</div>
`;

exports[`test bignav mount should test useEffect function 1`] = `0`;
